jobs/build-charms/charms.py

```diff
@@ -974,9 +976,16 @@ def build(
             entity.push()
             entity.attach_resources()
             entity.promote(to_channel=to_channel)
+        except Exception as build_exc:
+            failed_entities.append((entity, build_exc))
```

Comment:
I recommend something like `entity.echo(traceback.format_exc())` here so we get tracebacks for all the exceptions, not just the last one.